app/routes/[_]actions.newsletter.tsx

```diff
@@ -9,14 +9,14 @@ export const action = async ({ request }: ActionFunctionArgs) => {
   let body = new URLSearchParams(await request.text());
   let email = body.get("email");
   if (typeof email !== "string" || email.indexOf("@") === -1) {
-    return json({ error: "Invalid Email", ok: false }, { status: 400 });
+    return data({ error: "Invalid Email", ok: false }, { status: 400 });
   }
 
   try {
     await subscribeToNewsletter(email);
   } catch (e: any) {
-    return json({ error: e.message || "Unknown error", ok: false });
+    return data({ error: e.message || "Unknown error", ok: false });
   }
 
-  return json({ error: null, ok: true });
+  return data({ error: null, ok: true });
```

Comment:
This and the prior one don't _need_ `data()` because they don't care about the `status` but there's a known issue in our type inference that doesn't like the mixing and matching of data/json/etc. and raw objects so using data everywhere to avoid that.

Reply:
I left a comment on the [RR PR](https://github.com/remix-run/react-router-website/pull/118#discussion_r1757522208), should have read this one first.

Thanks for the explanation!